signac/contrib/job.py

```diff
@@ -510,35 +509,37 @@ def _init(self, force=False):
             (Default value = False).
 
         """
-        fn_manifest = os.path.join(self.workspace(), self.FN_MANIFEST)
-
         # Attempt early exit if the manifest exists and is valid
         try:
             statepoint = self._check_manifest()
         except Exception:
             # Any exception means this method cannot exit early.
 
-            # Create the workspace directory if it did not exist yet.
+            # Create the workspace directory if it does not exist.
             try:
                 _mkdir_p(self.workspace())
             except OSError:
                 logger.error(
                     "Error occurred while trying to create "
-                    "workspace directory for job '{}'.".format(self)
+                    "workspace directory for job '{}'.".format(self.id)
                 )
                 raise
 
+            fn_manifest = os.path.join(self.workspace(), self.FN_MANIFEST)
             try:
-                # Prepare the data before file creation and writing
-                blob = json.dumps(self.statepoint, indent=2)
+                # Prepare the data before file creation and writing.
+                statepoint = self.statepoint()
+                blob = json.dumps(statepoint, indent=2)
+            except JobsCorruptedError:
+                raise
 
-                try:
-                    # Open the file for writing only if it does not exist yet.
-                    with open(fn_manifest, "w" if force else "x") as file:
-                        file.write(blob)
-                except OSError as error:
-                    if error.errno not in (errno.EEXIST, errno.EACCES):
-                        raise
+            try:
+                # Open the file for writing only if it does not exist yet.
+                with open(fn_manifest, "w" if force else "x") as file:
+                    file.write(blob)
+            except OSError as error:
+                if error.errno not in (errno.EEXIST, errno.EACCES):
+                    raise
             except Exception as error:
                 # Attempt to delete the file on error, to prevent corruption.
```

Comment:
When you added:

https://github.com/glotzerlab/signac/blob/09a6677a963185634d1060b9495f6001bdadf4d7/signac/contrib/job.py#L529-L534

Did that prevent this portion of the code from erasing the manifest? - Just want to make sure I understand these changes properly.

Reply:
Yes. It will raise the `JobsCorruptedError` and will not continue to the later part of the code that attempts to roll back (delete) the manifest file creation.

Reply:
Thank you for the clarification!